internal/provider/resource_cloudflare_tunnel_config_test.go

```diff
@@ -0,0 +1,164 @@
+package provider
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func testTunnelConfig(resourceID, accountID, tunnelSecret string) string {
+	return fmt.Sprintf(`
+		resource "cloudflare_argo_tunnel" "%[1]s" {
+		  account_id = "%[2]s"
+		  name       = "%[1]s"
+		  secret     = "%[3]s"
+		}
+		
+		resource "cloudflare_tunnel_config" "%[1]s" {
+		  account_id         = "%[2]s"
+		  tunnel_id          = cloudflare_argo_tunnel.%[1]s.id
+		
+		  config {
+			warp_routing {
+			  enabled = true
+			}
+			origin_request {
+			  connect_timeout = "1m0s"
+			  tls_timeout = "1m0s"
+			  tcp_keep_alive = "1m0s"
+			  no_happy_eyeballs = false
+			  keep_alive_connections = 1024
+			  keep_alive_timeout = "1m0s"
+			  http_host_header = "baz"
+			  origin_server_name = "foobar"
+			  ca_pool = "/path/to/unsigned/ca/pool"
+			  no_tls_verify = false
+			  disable_chunked_encoding = false
+			  bastion_mode = false
+			  proxy_address = "127.0.0.1"
+			  proxy_port = "8123"
+			  proxy_type = "socks"
+			  ip_rules {
+				prefix = "/web"
+				ports = [80, 443]
+				allow = false
+			  }
+			}
+			ingress_rule {
+			  hostname = "foo"
+			  path = "/bar"
+			  service = "http://127.0.0.1:8080"
+			}
+			ingress_rule {
+				service = "https://127.0.0.1:8081"
+			  }
+		  }
+		}
+		`, resourceID, accountID, tunnelSecret)
+}
+
+func testTunnelConfigShort(resourceID, accountID, tunnelSecret string) string {
+	return fmt.Sprintf(`
+		resource "cloudflare_argo_tunnel" "%[1]s" {
+		  account_id = "%[2]s"
+		  name       = "%[1]s"
+		  secret     = "%[3]s"
+		}
+		
+		resource "cloudflare_tunnel_config" "%[1]s" {
+		  account_id         = "%[2]s"
+		  tunnel_id          = cloudflare_argo_tunnel.%[1]s.id
+		
+		  config {
+			warp_routing {}
```

Comment:
can we omit these fields if they aren't used? this is kinda unexpected for resources to need this type of setup.

Reply:
I do agree that it looks off, but I was running into issues with the defaults being on the empty return. All the `*time.Duration` fields throw a null reference exception because there is nothing on the response. It is the same for int, uint and bool as it is all pointers and using the `cloudflare.<Bool|Int|Uint>()` will return false, or 0 which fails the Acceptance tests as it is expecting nil. When having these weird empty values, the defaults get set and thus avoid this issue.